### PR TITLE
Added PDF in portal api

### PR DESCRIPTION
**Pull request by rskrzypczak**

### CODE REVIEW

File: api/webservice/Portal/BaseModule/Pdf.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Pdf file.
+ *
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/Pdf.php]: @package Api

File: api/webservice/Portal/BaseModule/Pdf.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	 * Check permission to method.
+	 *
+	 * @throws \Api\Core\Exception
+	 *
+	 * @return bool
+	 */
+	public function checkPermission()
+	{
+		$result = parent::checkPermission();
+		$moduleName = $this->controller->request->getModule('module');
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/Pdf.php]:
```suggestion
		$moduleName = $this->controller->request->getModule();
```

File: api/webservice/Portal/BaseModule/Pdf.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	 * Get method.
+	 *
+	 * @return array
+	 */
+	public function get()
+	{
+		$recordId = $this->controller->request->getInteger('record');
+		$templates = $this->controller->request->getArray('templates');
+		$file = $pdfFiles = $increment = [];
+		foreach ($templates as $templateId) {
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/Pdf.php]:
```suggestion
		foreach ($this->controller->request->getArray('templates') as $templateId) {
```

File: api/webservice/Portal/BaseModule/Pdf.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+	/**
+	 * Get method.
+	 *
+	 * @return array
+	 */
+	public function get()
+	{
+		$recordId = $this->controller->request->getInteger('record');
+		$templates = $this->controller->request->getArray('templates');
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/Pdf.php]:
```suggestion
```

File: api/webservice/Portal/BaseModule/PdfTemplates.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Pdf templates info.
+ *
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/PdfTemplates.php]: @Package Api

File: api/webservice/Portal/BaseModule/PdfTemplates.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	 * Check permission to method.
+	 *
+	 * @throws \Api\Core\Exception
+	 *
+	 * @return bool
+	 */
+	public function checkPermission()
+	{
+		$result = parent::checkPermission();
+		$moduleName = $this->controller->request->getModule('module');
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/PdfTemplates.php]:
```suggestion
		$moduleName = $this->controller->request->getModule();
```

File: api/webservice/Portal/BaseModule/PdfTemplates.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	/**
+	 * Get method.
+	 *
+	 * @return array
+	 */
+	public function get()
+	{
+		$moduleName = $this->controller->request->getModule('module');
```
Comment (mariuszkrzaczkowski) [inline api/webservice/Portal/BaseModule/PdfTemplates.php]:
```suggestion
		$moduleName = $this->controller->request->getModule();
```

File: modules/Vtiger/models/PDF.php [hunk 8]
```diff
@@ -571,4 +571,18 @@ public static function zipAndDownload(array $fileNames)
...
+	/**
+	 * Gets path.
+	 *
+	 * @param string $prefix
+	 *
+	 * @return string
+	 */
+	public function getPath(string $prefix = '')
+	{
+		$filePath = 'cache' . \DIRECTORY_SEPARATOR . 'pdf' . \DIRECTORY_SEPARATOR;
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/models/PDF.php]: ROOT_DIRECTORY ?

File: api/webservice/Portal/BaseModule/Pdf.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+	/**
+	 * Get method.
+	 *
+	 * @return array
+	 */
+	public function get()
+	{
+		$file = $pdfFiles = $increment = [];
+		$recordId = $this->controller->request->getInteger('record');
+		foreach ($this->controller->request->getArray('templates') as $templateId) {
```
Comment (TomaszQr) [inline api/webservice/Portal/BaseModule/Pdf.php]: $this->controller->request->getArray('templates', 'Integer')

**Pull request merged**